go/adbc/driver/bigquery/connection.go

```diff
@@ -0,0 +1,868 @@
+// Licensed to the Apache Software Foundation (ASF) under one
+// or more contributor license agreements.  See the NOTICE file
+// distributed with this work for additional information
+// regarding copyright ownership.  The ASF licenses this file
+// to you under the Apache License, Version 2.0 (the
+// "License"); you may not use this file except in compliance
+// with the License.  You may obtain a copy of the License at
+//
+//   http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing,
+// software distributed under the License is distributed on an
+// "AS IS" BASIS, WITHOUT WARRANTIES OR CONDITIONS OF ANY
+// KIND, either express or implied.  See the License for the
+// specific language governing permissions and limitations
+// under the License.
+
+package bigquery
+
+import (
+	"bytes"
+	"context"
+	"crypto/tls"
+	"encoding/json"
+	"errors"
+	"fmt"
+	"io"
+	"log"
+	"net/http"
+	"net/url"
+	"regexp"
+	"strconv"
+	"strings"
+	"time"
+
+	"golang.org/x/oauth2"
+
+	"cloud.google.com/go/bigquery"
+	"github.com/apache/arrow-adbc/go/adbc"
+	"github.com/apache/arrow-adbc/go/adbc/driver/internal"
+	"github.com/apache/arrow-adbc/go/adbc/driver/internal/driverbase"
+	"github.com/apache/arrow/go/v16/arrow"
+	"golang.org/x/exp/slices"
+	"google.golang.org/api/iterator"
+	"google.golang.org/api/option"
+)
+
+type connectionImpl struct {
+	driverbase.ConnectionImplBase
+
+	authType     string
+	credentials  string
+	clientID     string
+	clientSecret string
+	refreshToken string
+	projectID    string
+	datasetID    string
+	tableID      string
+
+	catalog  string
+	dbSchema string
+
+	client *bigquery.Client
+}
+
+type BigQueryTokenResponse struct {
```

Comment:
Do we need to be exporting this? Can we keep it internal? Along with the below handlers

Reply:
Sorry I forgot to make these private before I commit them.